### PR TITLE
Fix in unit tests and OTA agent

### PR DESCRIPTION
**Pull request by tgsong**
*Description of changes:*

OTA agent:
- Initialize pal callbacks
- Fix timer in OTA suspend

Unit tests:
- OTA agent
- Job parsing
- OS POSIX timers

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/unit-test/ota_os_posix_utest.c [hunk 1]
```diff
@@ -172,21 +171,21 @@ void test_OTA_posix_InvalidTimerOperations( void )
...
 
     /* Set the timeout to 0 and stop the timer*/
-    result = timer.start( timer.PTimerCtx, TIMER_NAME, 0, NULL );
+    result = timer.start( timer_id, TIMER_NAME, 0, NULL );
     TEST_ASSERT_EQUAL( OTA_ERR_NONE, result );
 
-    result = timer.stop( timer.PTimerCtx );
-    TEST_ASSERT_NOT_EQUAL( OTA_ERR_NONE, result );
+    result = timer.stop( timer_id );
+    TEST_ASSERT_EQUAL( OTA_ERR_NONE, result );
```
Comment (divekarshubham) [inline test/unit-test/ota_os_posix_utest.c:182]: Can we move this before we start a timer and expect to fail the timer stop?   TEST_ASSERT_NOT_EQUAL( OTA_ERR_NONE, result );

File: test/unit-test/ota_os_posix_utest.c [hunk 2]
```diff
@@ -153,15 +152,15 @@ void test_OTA_posix_TimerCreateAndStop( void )
...
     TEST_ASSERT_NOT_EQUAL( 0, getTimeElapsed() );
 
-    result = timer.stop( timer.PTimerCtx );
+    result = timer.stop( timer_id );
     TEST_ASSERT_EQUAL( OTA_ERR_NONE, result );
 
-    result = timer.delete( timer.PTimerCtx );
+    result = timer.delete( timer_id );
     TEST_ASSERT_EQUAL( OTA_ERR_NONE, result );
 }
```
Comment (divekarshubham) [inline test/unit-test/ota_os_posix_utest.c:166]: Doxygen briefs are interchanged for test_OTA_posix_TimerCreateAndStop() and test_OTA_posix_InvalidTimerOperations()

File: test/unit-test/ota_os_posix_utest.c [hunk 3]
```diff
@@ -172,21 +171,21 @@ void test_OTA_posix_InvalidTimerOperations( void )
 {
     OtaErr_t result = OTA_ERR_UNINITIALIZED;
 
-    result = timer.start( timer.PTimerCtx, TIMER_NAME, OTA_DEFAULT_TIMEOUT, NULL );
+    result = timer.start( timer_id, TIMER_NAME, OTA_DEFAULT_TIMEOUT, NULL );
     TEST_ASSERT_EQUAL( OTA_ERR_NONE, result );
 
     /* Set the timeout to 0 and stop the timer*/
-    result = timer.start( timer.PTimerCtx, TIMER_NAME, 0, NULL );
+    result = timer.start( timer_id, TIMER_NAME, 0, NULL );
```
Comment (divekarshubham) [inline test/unit-test/ota_os_posix_utest.c:178]: Can we pass an incorrect timer_id to fail the start?

**Review by divekarshubham — APPROVED** [review]

File: test/unit-test/ota_os_posix_utest.c (hunk 3 above)
Comment (tgsong) [reply test/unit-test/ota_os_posix_utest.c:178]: This pr is to fix the tests first, we can add more tests for failing path later.

File: test/unit-test/ota_os_posix_utest.c (hunk 1 above)
Comment (tgsong) [reply test/unit-test/ota_os_posix_utest.c:182]: This pr is to fix the tests first, we can add more tests for failing path later.

**Review by divekarshubham — APPROVED** [review]

**Review by yanjos-dev — APPROVED** [review]

**Pull request merged**